treeherder/perf/perf_sheriff_bot.py

```diff
@@ -15,14 +23,133 @@ class PerfSheriffBot:
     Wrapper class used to aggregate the reporting of backfill reports.
     """
 
-    def __init__(self, report_maintainer):
+    DEFAULT_MAX_RUNTIME = timedelta(minutes=50)
+
+    def __init__(
+        self,
+        report_maintainer: BackfillReportMaintainer,
+        backfill_tool: BackfillTool,
+        secretary_tool: SecretaryTool,
+        max_runtime: timedelta = None,
+        backfill_tool_disabled: bool = True,
+        secretary_tool_disabled: bool = True,
+    ):
         self.report_maintainer = report_maintainer
+        self.backfill_tool = backfill_tool
+        self.secretary = secretary_tool
+        self._woke_up_time = datetime.now()
```

Comment:
Why not assign the variables here instead of setting default args in __init__?

Reply:
I needed something that has these traits:
* all `PerfSheriffBot` instantiations in production code remain the same. Here `backfill_tool` & `secretary_tool` are completely disabled.
* provide an explicit way for enabling `backfill_tool` & `secretary_tool`, for testing purposes.
* this explicit way should be a bit annoying to do, so developers have some hints as to why is that *(it's not ready yet for production)*
* removing these feature flags should be straight forward. Just delete the marked source code.

Reply:
Right, I understand what you're saying about `backfill_tool_disabled` and `secretary_tool_disabled` but what I thought was redundant was this part: 

```
def __init__(
        self,
        report_maintainer: BackfillReportMaintainer,
        backfill_tool: BackfillTool,
        secretary_tool: SecretaryTool,
    ):
        self.report_maintainer = report_maintainer
        self.backfill_tool = backfill_tool
        self.secretary = secretary_tool
```

Unless you are anticipating different tools/maintainer classes to be used?